src/gui/dialogs/logging.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ Part of the Battle for Wesnoth Project http://www.wesnoth.org/
+
+ This program is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2 of the License, or
+ (at your option) any later version.
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY.
+
+ See the COPYING file for more details.
+ */
+
+#define GETTEXT_DOMAIN "wesnoth-lib"
+
+#include "gui/dialogs/logging.hpp"
+
+#include "gui/auxiliary/find_widget.hpp"
+#include "gui/widgets/grid.hpp"
+#include "gui/widgets/listbox.hpp"
+#include "gui/widgets/settings.hpp"
+#include "gui/widgets/toggle_button.hpp"
+#include "gui/widgets/window.hpp"
+
+#include "log.hpp"
+
+namespace gui2
+{
+
+REGISTER_DIALOG(logging)
+
+tlogging::tlogging()
+{
+    //list of names must match those in logging.cfg
+    widget_id.push_back("err");
+    widget_id.push_back("warn");
+    widget_id.push_back("info");
+    widget_id.push_back("debug");
+
+    domain_list_end = "the end";
+
+    //empty string is the filter (in other words, this grabs the whole list of domains)
+    std::string temp_string = lg::list_logdomains("");
+    //std::cout<<temp_string;//use to print the full log domain list
+    std::string one_domain;
+
+    std::istringstream iss(temp_string, std::istringstream::in);
+
+    while(iss >> one_domain)
+    {
+        domain_list.push_back(one_domain);
+    }
+    domain_list.push_back(domain_list_end);
+}
+
+void tlogging::pre_show(twindow& window)
+{
+    set_restore(true);//why is this done manually?
+
+	tlistbox& logger_box = find_widget<tlistbox>(&window, "logger_listbox", false);
+
+    for(unsigned int counter = 0; domain_list[counter] != domain_list_end; counter++)
+    {
+        std::string this_domain = domain_list[counter];
+		std::map<std::string, string_map> data;
+		string_map item;
+
+        for(unsigned int iter = 3; iter < 4; iter--)
+        {//counting down so that order matches logging.cfg
+            item["toggle_button"] = widget_id[iter];// id?
+            data.insert(std::make_pair(widget_id[iter], item));
+        }
+
+        /*to prevent all the individual toggles copying the label,
+         *(which will wreck everything)
+         *this widget MUST come last... I have no clue why
+         */
+		item["label"] = this_domain;
+		data.insert(std::make_pair("title_name", item));
```

Comment:
`data["title_name"] = item` seems cleaner to me in this situation. I'd use insert only if I already have a pair from somewhere else.